### PR TITLE
Introduce vulture to pre-commit hook

### LINKED ISSUE #197: Dead code analysis
Use https://github.com/jendrikseipp/vulture

- [x] check pre-commit hook
- [x] wire in to GitHub actions

### PR DESCRIPTION
**Pull request by kukovecz**
Fixes #197

### CODE REVIEW

**Review by QKaiser — APPROVED** [review]
I reviewed the dead code removal applied to handlers, it's great. Just wait for approval from the others on the vulture integration.

**Review by vlaci — APPROVED** [review]

**Pull request merged**
